src/mqtt_client/mqtt_client.rs

```diff
@@ -48,16 +48,25 @@ mod sm {
     statemachine! {
         transitions: {
             *Initial + Connected = ConnectedToBroker,
-            ConnectedToBroker + Subscribed = IndicatingLife,
-            IndicatingLife + IndicatedAlive / start_republish_timeout = PendingRepublish,
+            ConnectedToBroker + IndicatedLife = PendingSubscribe,
+
+            // After initial subscriptions, we start a timeout to republish all settings that is
+            // reset by any incoming updates.
```

Comment:
I would not reset the timeout on incoming updates. That creates unnecessary dependencies.